ENH: Prototype remote script download/execute

ericdill opened the pull request:
This adds a prototype implementation of detecting that the input to conda-execute
is actually a url and then downloading the contents of that url and executing it.

Addresses #18

ericdill commented:
Sort of hacky and almost certainly needs some exception handling around the usage of requests, but thoughts @pelson ?

pelson commented:
Ha! Neat. Thanks for doing this @ericdill.

I would probably break the input argument into protocol scheme and hierarchical part (i.e. ```['http', 'raw.githubusercontent.com/pelson/conda-execute/master/example.sh']``` and figure out which scheme we have (defaulting to file if undefined).

Otherwise, I don't think it is particularly hacky actually - I really like the simplicity of it.

pelson commented on conda_execute/execute.py:
```diff
@@ -158,6 +195,8 @@ def __call__(self, parser, namespace, values, option_string=None):
 
     try:
+        if args.path is not None and is_url(args.path):
+            args.code = download(args.path)
```
I would be tempted to just inline this whole function though. i.e.

```
args.code = download(args.path)
```

becomes

```
args.code = requests.get(args.path).content.decode()
```

ericdill replied:
Fair enough.  tbh I had made this its own function because I assumed that it was going to be more complicated than a one-liner to download a remote script :grinning:

ericdill commented:
Is this just a split on `://`?

```
['http', 'raw.githubusercontent.com/pelson/conda-execute/master/example.sh']
```

ericdill commented:
@pelson I rewrote this to do as you suggested and split the input argument on `://` which can be seen in this changeset: https://github.com/pelson/conda-execute/compare/master...ericdill:1c5dc2e.  

That looked a little overcomplicated to me, so I wanted to see what it looked like if I just left the check to see if `args.path` started with `http`.  That change is reflected as the current state of this PR.  I think I prefer the simpler implementation that does not split on `://`, but will defer to you.  Which one do you like more?

pelson commented:
LGTM! :+1: 
Thanks @ericdill
